packages/ui/src/ui/store/selectors/operations/operation.ts

```diff
@@ -31,7 +33,41 @@ interface OperationTask {
     task_name: string;
 }
 
-const getOperationAlertEvents = (state: RootState) => state.operations.detail.details.alert_events;
+const getOperationAlertEventsImpl = (state: RootState) =>
+    state.operations.detail.details.alert_events;
+
+export const getOperationAlertEvents = createSelector([getOperationAlertEventsImpl], (items) => {
+    const appeared: Record<string, AlertInfo> = {};
+    return reduce_(
+        items,
+        (acc, item) => {
+            const type = ypath.getValue(item.alert_type);
+            const code = ypath.getNumberDeprecated(item, '/error/code', NaN);
+            if (!code && appeared[type]) {
+                const last = appeared[type];
+                last.to = ypath.getValue(item.time);
+                delete appeared[type];
+            } else if (code) {
+                acc.push({
+                    from: ypath.getValue(item.time),
+                    type,
+                    error: item.error,
+                    url: prepareFaqUrl(type),
+                });
+                appeared[type] = acc[acc.length - 1];
+            } else {
+                acc.push({
+                    to: ypath.getValue(item.time),
+                    type,
+                    error: item.error,
+                    url: prepareFaqUrl(type),
+                });
+            }
+            return acc;
```

Comment:
I would not recommend the `if {} else if {} else {}` construction here. It complicates the readability of the code. I would just do `return acc` after the logical block:
if(code) {...; return acc;}
if(appeared[type]) {...; return acc;}
.... return acc;

This is not a mistake, so it's up to you.